### PR TITLE
Stages as network

### PR DESCRIPTION
**Pull request by schroedtert**

### CODE REVIEW

File: libsimulator/src/StageDescription.hpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright © 2012-2023 Forschungszentrum Jülich GmbH
+// Copyright © 2012-2023 Forechungszentrum Jülich GmbH
```
Comment (schroedtert) [inline libsimulator/src/StageDescription.hpp:1]: Correct!

File: libsimulator/src/VelocityModel.cpp [hunk 2]
```diff
@@ -187,7 +187,7 @@ double VelocityModel::GetSpacing(
     const GenericAgent& ped2,
     const Point direction) const
 {
-    assert(direction.IsUnitLength());
+    // assert(direction.IsUnitLength());
```
Comment (schroedtert) [inline libsimulator/src/VelocityModel.cpp:190]: Remove

File: libsimulator/test/TestJourney.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+TEST(RoundRobinTransition, ZeroWeightGivesException)
+{
+    int stage1;
+    std::vector<std::tuple<BaseStage*, uint64_t>> weightedStages = {
+        {reinterpret_cast<BaseStage*>(&stage1), 0}};
+
+    ASSERT_THROW(RoundRobinTransition sut(weightedStages), SimulationError);
+}
```
Comment (schroedtert) [inline libsimulator/test/TestJourney.cpp:70]: New line

File: libjupedsim/include/jupedsim/jupedsim.h [hunk 4]
```diff
@@ -397,6 +397,39 @@ JPS_GeometryBuilder_Build(JPS_GeometryBuilder handle, JPS_ErrorMessage* errorMes
...
 
+/**
+ * Opaque type for transition, describing route based decisions.
+ */
+typedef struct JPS_Transition_t* JPS_Transition;
+
+/**
+ * Create a fixed transition to stage
+ * @param[out] errorMessage if not NULL: will be set to a JPS_ErrorMessage in case of an error.
+ * @return Fixed transition to stage
```
Comment (Ozaq) [inline libjupedsim/include/jupedsim/jupedsim.h:408]: Need to mention that `NULL` will be returned on error.

File: libjupedsim/include/jupedsim/jupedsim.h [hunk 5]
```diff
@@ -397,6 +397,39 @@ JPS_GeometryBuilder_Build(JPS_GeometryBuilder handle, JPS_ErrorMessage* errorMes
...
+JUPEDSIM_API JPS_Transition
+JPS_Transition_CreateFixedTransition(JPS_StageId stageId, JPS_ErrorMessage* errorMessage);
+
+/**
+ * Create a round robin transition to stages
+ * @param stages target stages
+ * @param weights weights of target stage
+ * @param len length of stages and weights
+ * @param[out] errorMessage if not NULL: will be set to a JPS_ErrorMessage in case of an error.
+ * @return Round robin transition to stages
```
Comment (Ozaq) [inline libjupedsim/include/jupedsim/jupedsim.h:419]: Need to mention that NULL will be returned on error.

File: libjupedsim/include/jupedsim/jupedsim.h [hunk 6]
```diff
@@ -414,6 +447,19 @@ JUPEDSIM_API JPS_JourneyDescription JPS_JourneyDescription_Create();
  */
 JUPEDSIM_API void JPS_JourneyDescription_AddStage(JPS_JourneyDescription handle, JPS_StageId id);
 
+/**
+ * Specifies the transition to the next stage, once this stage is completed.
+ * @param handle of the JourneyDescription to modify.
+ * @param id of the stage to set the transition for.
+ * @param transition transition to the next stage.
+ * @param[out] errorMessage if not NULL: will be set to a JPS_ErrorMessage in case of an error.
+ */
```
Comment (Ozaq) [inline libjupedsim/include/jupedsim/jupedsim.h:456]: @return doc missing

Comment (schroedtert) [reply libjupedsim/include/jupedsim/jupedsim.h:457]: added

File: libsimulator/src/StageDescription.hpp (hunk 1 above)
Comment (schroedtert) [reply libsimulator/src/StageDescription.hpp]: changed

File: libsimulator/src/VelocityModel.cpp (hunk 2 above)
Comment (schroedtert) [reply libsimulator/src/VelocityModel.cpp]: removed

File: libjupedsim/include/jupedsim/jupedsim.h (hunk 4 above)
Comment (schroedtert) [reply libjupedsim/include/jupedsim/jupedsim.h]: Added remark

File: libjupedsim/include/jupedsim/jupedsim.h (hunk 5 above)
Comment (schroedtert) [reply libjupedsim/include/jupedsim/jupedsim.h]: Added remark

**Pull request merged**